src/robusta/integrations/slack/sender.py

```diff
@@ -213,16 +219,37 @@ def __to_slack(self, block: BaseBlock, sink_name: str) -> List[SlackBlock]:
             logging.warning(f"cannot convert block of type {type(block)} to slack format block: {block}")
             return []  # no reason to crash the entire report
 
-    def __upload_file_to_slack(self, block: FileBlock, max_log_file_limit_kb: int) -> str:
+    def __upload_file_to_slack(self, block: FileBlock, max_log_file_limit_kb: int) -> Optional[str]:
+        """Upload a file to Slack and return a permalink to it."""
         truncated_content = block.truncate_content(max_file_size_bytes=max_log_file_limit_kb * 1000)
+        filename = block.filename
 
-        """Upload a file to slack and return a link to it"""
-        with tempfile.NamedTemporaryFile() as f:
+        def _upload_temp_file(file_reference) -> Optional[str]:
+            """Helper to upload a file-like or file path to Slack."""
             f.write(truncated_content)
             f.flush()
-            result = self.slack_client.files_upload_v2(title=block.filename, file=f.name, filename=block.filename)
+            f.seek(0)
+
+            result = self.slack_client.files_upload_v2(
+                title=filename,
+                file_uploads=[{"file": file_reference, "filename": filename, "title": filename}],
+            )
             return result["file"]["permalink"]
 
+        try:
+            with tempfile.NamedTemporaryFile() as f:
+                logging.debug("Trying NamedTemporaryFile for Slack upload")
+                return _upload_temp_file(f.name)
```

Comment:
How this works? You are passing the name which is string. am i missing something?

Reply:
string of the location in the filesystem - full filepath

Reply:
so it uses f as inner function arg instead of passing it to the inner function? It just harder to understand what is going on here and why we have done it that way. Please move the inner function our and pass the file as arg.